src/serializer/utils.js

```diff
@@ -249,3 +253,101 @@ export function handleReportedSideEffect(
     exceptionHandler(`side-effects from throwing exception${location}`, expressionLocation);
   }
 }
+
+export class ResidualOptimizedFunctions {
```

Comment:
This thing is substantial. Put it in its own file, similar to all the other `ResidualXXX` classes.